app/src/main/java/io/appium/uiautomator2/model/UiAutomationElement.java

```diff
@@ -124,11 +127,7 @@ public static UiAutomationElement rebuildForNewRoot(AccessibilityNodeInfo rawEle
     }
 
     @Nullable
-    public static UiAutomationElement getCachedElement(AccessibilityNodeInfo rawElement,
-                                                       AccessibilityNodeInfo windowRoot) {
-        if (cache.get(rawElement) == null) {
-            rebuildForNewRoot(windowRoot, null);
-        }
+    public static UiAutomationElement getCachedElement(AccessibilityNodeInfo rawElement) {
```

Comment:
In how many places this call is used? Can it break anything else if changed this way?

Reply:
I'd. rather pass the array of window roots to keep the backward compatibility

Reply:
I saw it used in only one place. When I looked at it, I didn't see the point of having that logic in UiAutomationElement. I can revert.